### PR TITLE
Level10

### PR DESCRIPTION
**Pull request by ElizavetaSergeeva-Nekrasova**

### CODE REVIEW

File: src/ComparatorByDefectCount.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import java.util.Comparator;
+
+public class ComparatorByDefectCount implements Comparator<Vehicle> {
+    @Override
+    public int compare(Vehicle o1, Vehicle o2) {
+        return (MechanicService.getSumNumberOfBreaks(o2)) - (MechanicService.getSumNumberOfBreaks(o1));
```
Comment (AlexeyKatomachin) [inline src/ComparatorByDefectCount.java]: it's should be in similar style
in another class you comparing by Double.compare
maybe hear also better to compare by Integer.compare or something like that

Comment (ElizavetaSergeeva-Nekrasova) [reply src/ComparatorByDefectCount.java]: fixed

File: src/Main.java [hunk 2]
```diff
@@ -1,153 +1,63 @@
...
-                max = breaksCount;
-                vehicleWithMaxBreaks = vehicle;
-            }
-
-            mechanicService.repair(vehicle);
-        }
-
-        System.out.println("Car with the maximum number of breaks: " + vehicleWithMaxBreaks);
+    private static void showVehicles(List<Vehicle> vehicleList) {
+        vehicleList
```
Comment (nikoteressi) [inline src/Main.java]: stream is unnecessary here. You can call just forEach().

File: src/Main.java [hunk 3]
```diff
@@ -1,153 +1,63 @@
...
-        }
+    private static void repairVehicleList(List<Vehicle> vehicleList, MechanicService mechanicService) {
+        vehicleList
+                .stream()
+                .forEach(vehicle -> {
+                    if (!mechanicService.isBroken(vehicle)) {
+                        System.out.println("Auto " + vehicle.getId() + " исправно");
+                    } else {
+                        System.out.println("Auto " + vehicle.getId() + " неисправно, чиним...");
+                        mechanicService.repair(vehicle);
```
Comment (nikoteressi) [inline src/Main.java]: I believe you can move this logic into a separate method

File: src/RentVehicle.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+public class RentVehicle {
+    public static Rent rentVehicle(Vehicle vehicle, double cost,  MechanicService mechanicService)
+            throws DefectedVehicleException {
+        if (!mechanicService.isBroken(vehicle)) {
+            return new Rent(vehicle.getId(), new Date(), cost);
+        }
+
+        throw new DefectedVehicleException("Vehicle is defected");
+    }
```
Comment (nikoteressi) [inline src/RentVehicle.java]: public static Rent rentVehicle(Vehicle vehicle, double cost,  MechanicService mechanicService)
            throws DefectedVehicleException {
        if (mechanicService.isBroken(vehicle)) {
              throw new DefectedVehicleException("Vehicle is defected");
        }
        return new Rent(vehicle.getId(), new Date(), cost);
    }

such way looks better)

**Review by AlexeyKatomachin — APPROVED** [review]

File: src/Main.java (hunk 2 above)
Comment (ElizavetaSergeeva-Nekrasova) [reply src/Main.java]: fixed

File: src/Main.java (hunk 3 above)
Comment (ElizavetaSergeeva-Nekrasova) [reply src/Main.java]: fixed

File: src/RentVehicle.java (hunk 4 above)
Comment (ElizavetaSergeeva-Nekrasova) [reply src/RentVehicle.java]: Thanks, it's really better, fixed

**Review by nikoteressi — APPROVED** [review]

**Pull request merged**